gsl_projet/tests/test_views.py

```diff
@@ -295,10 +284,9 @@ def test_filter_by_dotation_dsil_and_detr_dsil(
     assert Projet.objects.count() == 11
 
     assert qs.count() == 2 + 4
-    for p in qs:
-        dotation = p.dossier_ds.demande_dispositif_sollicite
-        assert ("DSIL" in dotation and "DETR" in dotation) or "DSIL" in dotation
-        assert not ("DETR" in dotation and "DSIL" not in dotation)
+    assert qs.filter(dsil_count__gt=0).count() == 6
+    assert qs.filter(detr_count=0, dsil_count__gt=0).count() == 2
+    assert qs.filter(detr_count__gt=0).count() == 4
```

Comment:
Je suis un peu gênée que l'on teste ici le détail d'implémentation : si tu changes la façon dont ça marche, tu vas devoir changer le test, alors que l'interface ne change pas.

J'irais plutôt tester `projet.dotations`, quelque chose comme ça : 

```suggestion
    assert all("DSIL" in projet.dotations for projet in qs)
    assert sum(1 for projet in qs if "DETR" not in projet.dotations) == 2
    assert sum(1 for projet in qs if "DETR" in projet.dotations) == 4
```

Reply:
Cela étant dit, je trouve l'implémentation elle-même très élégante !

Reply:
Ah oui, ça peut-être bien !
La property `dotations` n'est pas dans cette PR, donc j'attends de merge la #194 avant de rebase ici et d'appliquer ces suggestions !